warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PushdownNegationVisitor.java

```diff
@@ -41,7 +39,10 @@ public static JexlNode pushdownNegations(JexlNode tree) {
         flattened.jjtAccept(new PushdownNegationVisitor(), new NegationState(false));
         
         // flatten the result
-        return TreeFlatteningRebuildingVisitor.flatten(flattened);
+        flattened = TreeFlatteningRebuildingVisitor.flatten(flattened);
+        
+        // remove extra parens and return
+        return RemoveExtraParensVisitor.remove(flattened);
```

Comment:
Can't we fix the TreeFlatteningRebuildingVisitor to remove extra parens up front?

Reply:
I could try rolling this into the TreeFlatteningRebuildingVisitor

Reply:
This approach is quickly proving to be too unwieldy.

Reply:
Alternatively can we add this to the TreeFlatteningRebuildingVisitor.flatten() even as its own visitor so that the expected result is at least the same all the time.